src/NUnitConsole/nunit3-console.tests/ResultReporterTests.cs

```diff
@@ -169,11 +177,23 @@ public void TestsNotRunTest()
             Assert.That(report, Is.EqualTo(expected));
         }
 
-        #region Helper Methods
+        [Test]
+        public void TestParameterSettingsWrittenCorrectly()
+        {
+            var expected = new[] {
+                "    TestParameters: 1=d;2=c",
+                "    TestParametersDictionary:",
+                "        1 -> d",
+                "        2 -> c",
+                "    NumberOfTestWorkers: 8"
+            };
 
-        private TestEngineResult AddMetadata(TestEngineResult input)
+            var report = GetReportLines(_reporter.WriteRunSettingsReport);
+            Assert.That(report, Is.SupersetOf(expected));
+        }
+
+        private static TestEngineResult AddMetadata(TestEngineResult input)
         {
-            input.Add("<settings><setting name=\"Setting1Name\" value=\"Setting1Value\"></setting><setting name=\"Setting2Name\" value=\"Setting2Value\"></setting></settings>");
```

Comment:
At least if we can trust https://github.com/nunit/docs/wiki/Test-Result-XML-Format

Reply:
Settings can appear in different places because some settings apply only to an assembly. The ones you are concerned with should actually be created at the engine level, at least to my memory, so I'm a bit surprised here. This is a more or less obscure part of the architecture (not well documented) so it could be that some things have moved around over time.

Reply:
I _think_ I had a look to see if there was any code to create `settings` at test-run level, and couldn't find any - but it was a while ago I did that bit now, so can't be sure!

Reply:
It looks as if all settings have moved to the assembly level, including those that function at a higher level than the assembly. So no problem. 😄